rest/config.go

```diff
@@ -1430,10 +1438,38 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 			cnf.KeyPath = sc.Config.Bootstrap.X509KeyPath
 		}
 		base.TracefCtx(ctx, base.KeyConfig, "Got config for bucket %q with cas %d", bucket, cas)
-		return true, &cnf, nil
+		return true, nil
 	}
 
-	return false, nil, nil
+	err = sc.foreachDbConfig(callback)
+	if err != nil {
+		return false, nil, nil
+	}
+
+	return true, &cnf, nil
+}
+
+func (sc *ServerContext) bucketNameFromDbName(dbName string) (bucketName string, found bool) {
+	// Only used in handleDeleteDB in config.go when GetDatabase does not find the db (CBG-2420)
```

Comment:
I usually find these types of comments get quickly out of date unless there's a very specific reason why this function should ONLY be used in one particular codepath. Probably best to just remove it. We can also trace it back to this PR with `git blame`/annotations

It might be worth adding a comment though, to explain why we're not using the `DatabaseConfig` struct and instead are creating a smaller one.

```suggestion
	// Minimal representation of config struct to be tolerant of invalid database configurations where we still need to find a database name
```